extensions/smallrye-metrics/deployment/src/main/java/io/quarkus/smallrye/metrics/deployment/SmallRyeMetricsProcessor.java

```diff
@@ -481,48 +480,6 @@ void registerRuntimeExtensionMetrics(
             List<MetricsFactoryConsumerBuildItem> metricsFactoryConsumerBuildItems,
             ValidationPhaseBuildItem validationPhase,
             BeanArchiveIndexBuildItem beanArchiveIndex) {
-        IndexView index = beanArchiveIndex.getIndex();
-        for (io.quarkus.arc.processor.BeanInfo bean : validationPhase.getContext().beans().producers()) {
-            ClassInfo implClazz = bean.getImplClazz();
-            if (implClazz == null) {
-                continue;
-            }
-            MetricType metricType = getMetricType(implClazz);
-            if (metricType != null) {
```

Comment:
Do you want to emit any kind of warning if use is detected? We do this for a few other things... Given it is a known breaking change for MP 4.0, not sure it is necessary (there also may already be a warning I'm forgetting about)..

Reply:
Good point, done that